azurerm/resource_arm_user_assigned_identity.go

```diff
@@ -0,0 +1,130 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/msi/mgmt/2015-08-31-preview/msi"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmUserAssignedIdentity() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmUserAssignedIdentityCreate,
+		Read:   resourceArmUserAssignedIdentityRead,
+		Update: resourceArmUserAssignedIdentityCreate,
+		Delete: resourceArmUserAssignedIdentityDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
I'd missed there wasn't an Import test for this (these are tracked in a separate file) - I've included this in the commit (but [here's an example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/import_arm_search_service_test.go#L10)); these verify the state set via `d.Set` matches the configuration used to create it (by creating the resource, then removing it from the state, then running `terraform import` and ensuring the config matches the state)